### PR TITLE
[UA update] chore: Update User-Agent string

### PR DESCRIPTION
**Pull request by 96malhar**
*Issue #, if available:*
SDK-169882

*Description of changes:*
chore: Update User-Agent string.

*Testing*
Verified that the updated UA string is `lib/aws-dotnet-deploy-cli#1.20.3.0 md/cli-args#{cli_command} `
* The revision part of the version number is intentionally set to 0 since versioning on NuGet follows semantic versioning consisting of `Major.Minor.Patch` versions.
* The `cli_command` can be one of `deploy`, `list-deployments`, `delete-deployment`, `deployment-project`, `server-mode`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj [hunk 1]
```diff
@@ -21,11 +21,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.IdentityManagement" Version="3.7.200.42" />
-    <PackageReference Include="AWSSDK.CloudFormation" Version="3.7.203.36" />
-    <PackageReference Include="AWSSDK.SecurityToken" Version="3.7.201.38" />
-    <PackageReference Include="AWSSDK.SSO" Version="3.7.200.42" />
-    <PackageReference Include="AWSSDK.SSOOIDC" Version="3.7.200.42" />
+    <PackageReference Include="AWSSDK.IdentityManagement" Version="3.7.301.3" />
```
Comment (ashovlin) [inline src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:24]: Can you update the internal tracker and use that to regenerate https://github.com/aws/aws-dotnet-deploy/blob/main/THIRD_PARTY_LICENSES?

File: src/AWS.Deploy.CLI/App.cs [hunk 2]
```diff
@@ -48,29 +50,48 @@ public async Task<int> Run(string[] args)
...
+            var version = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version;
+            if (version is null)
             {
-                envValue.Append($"_{args[0]}");
+                return string.Empty;
             }
 
-            Environment.SetEnvironmentVariable(envName, envValue.ToString());
+            var versionParts = version.Split('.');
+            if (versionParts.Length == 4)
```
Comment (ashovlin) [inline src/AWS.Deploy.CLI/App.cs:89]: Could you add a comment about why you're doing this?

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj (hunk 1 above)
Comment (96malhar) [reply src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:24]: Done

File: src/AWS.Deploy.CLI/App.cs (hunk 2 above)
Comment (96malhar) [reply src/AWS.Deploy.CLI/App.cs:86]: Done

**Review by normj — APPROVED** [review]

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj (hunk 1 above)
Comment (ashovlin) [reply src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:24]: Did you update the internal tool? I still see AWSSDK.IdentityManagement at 3.7.200.42?

Comment (96malhar) [reply src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:24]: Just updated the internal tool as well.

**Review by ashovlin — APPROVED** [review]

**Pull request merged**